native/cocos/platform/android/java/vendor/google/billing/GoogleBilling.java

```diff
@@ -28,84 +28,112 @@ of this software and associated documentation files (the "Software"), to deal
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
-import com.android.billingclient.api.BillingConfig;
+
 import com.android.billingclient.api.BillingResult;
-import com.android.billingclient.api.InAppMessageResult;
 import com.android.billingclient.api.ProductDetails;
 import com.android.billingclient.api.Purchase;
 import com.android.billingclient.api.AcknowledgePurchaseParams;
 import com.android.billingclient.api.AcknowledgePurchaseResponseListener;
 import com.android.billingclient.api.BillingClient;
-import com.android.billingclient.api.BillingClientStateListener;
+
 import com.android.billingclient.api.BillingFlowParams;
 import com.android.billingclient.api.ConsumeParams;
 import com.android.billingclient.api.ConsumeResponseListener;
 import com.android.billingclient.api.AlternativeBillingOnlyReportingDetailsListener;
 import com.android.billingclient.api.ProductDetailsResponseListener;
 import com.android.billingclient.api.PurchasesResponseListener;
-import com.android.billingclient.api.PurchasesUpdatedListener;
 import com.android.billingclient.api.QueryProductDetailsParams;
 import com.android.billingclient.api.QueryPurchasesParams;
 import com.android.billingclient.api.GetBillingConfigParams;
 import com.android.billingclient.api.BillingConfigResponseListener;
-import com.android.billingclient.api.AlternativeBillingOnlyReportingDetails;
 import com.android.billingclient.api.AlternativeBillingOnlyAvailabilityListener;
 import com.android.billingclient.api.AlternativeBillingOnlyInformationDialogListener;
 import com.android.billingclient.api.ExternalOfferReportingDetailsListener;
-import com.android.billingclient.api.ExternalOfferReportingDetails;
 import com.android.billingclient.api.ExternalOfferAvailabilityListener;
 import com.android.billingclient.api.ExternalOfferInformationDialogListener;
 import com.android.billingclient.api.InAppMessageParams;
 import com.android.billingclient.api.InAppMessageResponseListener;
+import com.android.billingclient.api.BillingClientStateListener;
 import com.cocos.lib.GlobalObject;
-import com.cocos.lib.CocosHelper;
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class GoogleBilling implements BillingClientStateListener,
-    PurchasesUpdatedListener, ProductDetailsResponseListener, PurchasesResponseListener,
-    ConsumeResponseListener, AcknowledgePurchaseResponseListener, BillingConfigResponseListener,
-    AlternativeBillingOnlyReportingDetailsListener, AlternativeBillingOnlyAvailabilityListener,AlternativeBillingOnlyInformationDialogListener,
-    ExternalOfferReportingDetailsListener, ExternalOfferAvailabilityListener,ExternalOfferInformationDialogListener, InAppMessageResponseListener {
+public class GoogleBilling {
```

Comment:
If there are no subclasses, it is recommended to declare it as a final class.

Reply:
fixed.